tests/mfem_ex9p_blockilu.cpp

```diff
@@ -0,0 +1,758 @@
+    }
+    dc->RegisterField("solution", u);
+    dc->SetCycle(0);
+    dc->SetTime(0.0);
+    dc->Save();
+  }
+
+  ParaViewDataCollection* pd = NULL;
+  if (paraview) {
+    pd = new ParaViewDataCollection("Example9P", pmesh);
+    pd->SetPrefixPath("ParaView");
+    pd->RegisterField("solution", u);
+    pd->SetLevelsOfDetail(order);
+    pd->SetDataFormat(VTKFormat::BINARY);
+    pd->SetHighOrderOutput(true);
+    pd->SetCycle(0);
+    pd->SetTime(0.0);
+    pd->Save();
+  }
+
+  // Optionally output a BP (binary pack) file using ADIOS2. This can be
+  // visualized with the ParaView VTX reader.
+#ifdef MFEM_USE_ADIOS2
+  ADIOS2DataCollection* adios2_dc = NULL;
+  if (adios2) {
+    std::string postfix(mesh_file);
+    postfix.erase(0, std::string("../data/").size());
+    postfix += "_o" + std::to_string(order);
+    const std::string collection_name = "ex9-p-" + postfix + ".bp";
+
+    adios2_dc = new ADIOS2DataCollection(MPI_COMM_WORLD, collection_name, pmesh);
+    // output data substreams are half the number of mpi processes
+    adios2_dc->SetParameter("SubStreams", std::to_string(num_procs / 2));
+    // adios2_dc->SetLevelsOfDetail(2);
+    adios2_dc->RegisterField("solution", u);
+    adios2_dc->SetCycle(0);
+    adios2_dc->SetTime(0.0);
+    adios2_dc->Save();
+  }
+#endif
+
+  socketstream sout;
+  if (visualization) {
+    char vishost[] = "localhost";
+    int  visport   = 19916;
+    sout.open(vishost, visport);
+    if (!sout) {
+      if (myid == 0) cout << "Unable to connect to GLVis server at " << vishost << ':' << visport << endl;
+      visualization = false;
+      if (myid == 0) {
+        cout << "GLVis visualization disabled.\n";
+      }
+    } else {
+      sout << "parallel " << num_procs << " " << myid << "\n";
+      sout.precision(precision);
+      sout << "solution\n" << *pmesh << *u;
+      sout << "pause\n";
+      sout << flush;
+      if (myid == 0)
+        cout << "GLVis visualization paused."
+             << " Press space (in the GLVis window) to resume it.\n";
+    }
+  }
+
+  // 10. Define the time-dependent evolution operator describing the ODE
+  //     right-hand side, and perform time-integration (looping over the time
+  //     iterations, ti, with a time-step dt).
+  FE_Evolution adv(*m, *k, *B);
+
+  double t = 0.0;
+  adv.SetTime(t);
+  ode_solver->Init(adv);
+
+  bool done = false;
+  for (int ti = 0; !done;) {
+    double dt_real = min(dt, t_final - t);
+    ode_solver->Step(*U, t, dt_real);
+    ti++;
+
+    done = (t >= t_final - 1e-8 * dt);
+
+    if (done || ti % vis_steps == 0) {
+      if (myid == 0) {
+        cout << "time step: " << ti << ", time: " << t << endl;
+      }
+
+      // 11. Extract the parallel grid function corresponding to the finite
+      //     element approximation U (the local solution on each processor).
+      *u = *U;
+
+      if (myid == 0) {
+        EXPECT_FLOAT_EQ(u->Norml2(), UNMODIFIED_SOLN.at((ti / vis_steps) - 1));
+      }
```

Comment:
I also hardcoded the results from the unmodified version of the example and added a comparison.

Reply:
This should probably be moved to an integration test with a binary comparison (along with most of the current tests) once that capability comes online.

Reply:
Also moved to a more Serac way of programming at the same time but not now.. in the future